openassessment/workflow/models.py

```diff
@@ -201,30 +201,14 @@ def score(self):
             score = sub_api.get_latest_score_for_submission(self.submission_uuid)
         return score
 
-    def status_details(self, assessment_requirements):
+    def status_details(self):
         status_dict = {}
         steps = self._get_steps()
         for step in steps:
-            api = step.api()
-            if api is not None:
```

Comment:
This whole block is more-or-less duplicated by https://github.com/edx/edx-ora2/blob/efischer/training_reqs_take3/openassessment/workflow/models.py#L636, and both blocks are executed (but yield slightly different results!) when `api.update_from_assessments()` is called.

If `status_details()` is to remain a "update, then get details" method, it should call the step's `update()` method instead of duplicating the logic. In that case, we should also update `api.update_from_assessments()` to only call this method.

I personally prefer this solution, where `update()` (writes data) and `status_details()` (reads data) are separated.